slides/git-intro/index.html

```diff
@@ -0,0 +1,363 @@
+                    <p>Sometimes, you may need to make changes to a branch that hasn't yet been merged into master.</p>
+                    <p>Perhaps a pull request (...more later) needs modified, after you've started a new feature (on a new branch, of course....)</p>
+                    <p>In this situation, you can <strong>checkout</strong> the specific branch to work on</p>
+                    <pre>
+                        <code class="sh" data-trim contenteditable>
+                            $ git checkout hotfix_001
+                        </code>
+                    </pre>
+                </section>
+
+                <section>
+                    <h2>Staging</h2>
+                    <p>As you begin working on new features, you may find yourself adding new files to the repository.</p>
+                    <p>In this case, you are effectively 'staging' these files, before you <strong>commit</strong> (...more later) them to your branch.</p>
+                    <pre>
+                        <code class="zsh" data-trim contenteditable>
+                            $ git add 'filename'
+                        </code>
+                    </pre>
+                </section>
+
+                <section>
+                    <h2>Why Stage?</h2>
+                    <p>Staging allows you to selectively add certain files to your next commit, whilst leaving the rest out of the commit.</p>
+                    <p>You may have made changes in several files, but you only want one of those files to be added to the branch, and scrap the rest.</p>
+                    <p>In this scenario, you can stage the files you want to keep in the <strong>index</strong>, and throw away the rest after committing.</p>
+                </section>
+
+                <section>
+                    <h2>Git Commit</h2>
+                    <p>Once you're happy with the changes in your staging area, you commit them to your branch.</p>
+                    <p>Committing changes is a way of saying <em>"I'm happy with this, this can go to master someday..."</em>.</p>
+                    <p>Rolling back commits is a tedious process, so make sure you're 100% with the changes before committing!</p>
+                    <pre>
+                        <code class="zsh" data-trim contenteditable>
+                            $ git commit -m 'Adding feature to process x in the y system'
+                        </code>
+                    </pre>
+                </section>
+
+                <section>
+                    <h2>Git Commit Messages</h2>
+                    <p>Remember our Git History? Here's where it comes from. Every commit should have a descriptive message describing what the change is for.</p>
+                    <p>NB: <strong>Miscellaneous Changes</strong> is not a good message!</p>
+                    <p>The <strong>-m</strong> flag when using <strong>git commit</strong> tells Git that the commit message follows, so it won't prompt for it.</p>
```

Comment:
Just in interest of shortening the text, I think we can delete the last line here. The code sample before covers it well.

I really like the point youre making about good messages though. It might be nice to fire in a screenshot of a git log output or a Github commit page.

Obviously from a public repo, can use this one or https://github.com/DeloitteDigitalUK/react-redux-starter-app if something more dev related is helpful? (Neither is hugely careful about the messages, but meh!)